### ISSUE TITLE
[@angular-eslint/template/prefer-control-flow] false positive with [ngFormOptions]

### ISSUE THREAD

**Issue by mopinsk** [angular-eslint/angular-eslint#1633]
**Description and reproduction of the issue**
When using the rule **@angular-eslint/template/prefer-control-flow** with **[ngFormOptions]** I get the following error:

Use built-in control flow instead of directive ngFormOptions.

I guess this is because "**[ngFormOptions]**" starts with "**[ngFor**".

```JSON
{
  "extends": ["plugin:@angular-eslint/template/all"]
}
```

```HTML
<form [ngFormOptions]="{ updateOn: 'blur' }">
...
</form>
```

**Versions**

| package                                   | version |
| ----------------------------------        | ------- |
| `@angular-eslint/eslint-plugin`  | `17.1.0` |
| `@angular-eslint/eslint-plugin-template`  | `17.1.0` |
| `@angular-eslint/template-parser`         | `17.1.0` |
| `@typescript-eslint/eslint-plugin`               | `6.12.0` |
| `@typescript-eslint/parser`               | `6.12.0` |
| `ESLint`                                  | `8.54.0` |
| `node`                                    | `18.17.1` |

```sh
Angular CLI: 17.0.3
Node: 18.17.1
Package Manager: npm 9.8.1
OS: win32 x64

Angular: 17.0.4
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1700.3
@angular-devkit/build-angular   17.0.3
@angular-devkit/core            17.0.3
@angular-devkit/schematics      17.0.3
@angular/cdk                    17.0.1
@angular/cli                    17.0.3
@angular/material               17.0.1
@schematics/angular             17.0.3
rxjs                            7.8.1
typescript                      5.2.2
zone.js                         0.14.2
```

- [x] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by JamesHenry** [comment]
FYI @json-derulo
